frontend/app.py

```diff
@@ -16,7 +16,7 @@ def search(query: str, endpoint: str) -> dict:
     response = requests.post(endpoint, headers=headers, data=data)
     content = response.json()
 
-    matches = content["data"]["docs"]
+    matches = content["data"]["docs"][0]["matches"]
```

Comment:
Why this? Did you notice any problems with `content["data"]["docs"]` only?

Reply:
`docs.matches()` appears to change `docs` in place.

It returns an array with a single element which is a dict that has key _matches_ and contains the matches to the search query.

There is no longer accessible data at content["data"]["docs"]. Have a play with `content` and see what content looks like, the structure was a tad complicated but appending `[0]["matches"]` recovered what we had before.

One could also investigate what exactly `docs.matches` is doing to `docs`.

Reply:
Ok great, thanks for the explanation!